packages/chopsticks/src/blockchain/index.ts

```diff
@@ -105,12 +120,12 @@ export class Blockchain {
     if (number > this.#head.number) {
       return undefined
     }
-    if (!this.#blocksByNumber[number]) {
+    if (!this.#blocksByNumber.has(number)) {
       const hash = await this.api.getBlockHash(number)
```

Comment:
I didn't get from db in getBlockAt as the hash is not provided, if get from db by number then it could be reliable if there are duplicated blocks

Reply:
in that case we need a way to mark a block as canonical. one easy way is delete the old one. alternatively mark the new one as canonical and unmark the old canonical block